chapters/feature-selection.qmd

```diff
@@ -0,0 +1,525 @@
+Depending on the context, our interest in feature selection could be at either level. If the original predictors are expensive to estimate, we would be more interested in removing them at the original level; if _any_ of their derived predictors are important, we need the original. 
+
+This idea will arise again in @sec-importance when we discuss _variable importance scores_ for explaining models. 
+
+## Overfitting the Feature Set {#sec-selection-overfitting}
+
+We’ve previously discussed overfitting, where a model finds patterns in the training data that are not reproducible. It is also possible to _overfit the predictor set_ by finding the training set predictors that appear to be connected to the outcome but do not show the same relationship in other data. 
+
+This is most likely to occur when the number of predictors ($p$) is much larger than the training set size ($n_{tr}$) or when no external validation is used to verify performance. @Ambroise2002p1493 shows reanalyses of high-dimensional biology experiments using a backward selection algorithm for removing predictors. In some cases, they could find a model and predictor subset with perfect accuracy even when the training set outcome data were randomly shuffled. 
+
+Recall back in @sec-model-pipeline an argument was made that preprocessing methods _and_ the supervised model encompassed the entire model pipeline. Technically, feature selection (supervised or unsupervised) falls into the definition of preprocessing. It is important to use data that is different from the data used to train the pipeline (i.e., select the predictors) is different from the data used to evaluate how well it worked.  
+
+For example, if we use a resampling scheme, we have to repeat the feature selection process within each resample. If we used 10-fold cross-validation, each of the 10 analysis sets (90% of the training set) would have its own set of selected predictors. 
+
+@Ambroise2002p1493 demonstrated that when feature selection was enacted once, and then the model with this subset was resampled, they could produce models with excessively optimistic performance metrics. However, when the performance statistics were computed with multiple realizations of feature selection, there was very little false optimism. 
+
+Another improper example of data usage that can be found in the literature, as well as in real-life analyses, occurs when the entire data set is used to select predictors _then_ the initial split is used to create the training and test sets. This “bakes in” the false optimism from the start^[Recall out mantra that “Always have a separate piece of data that can **contradict what you believe**.”]. 
+
+However, performing feature selection in each resample can drastically increase the computational cost, especially if the model has many tuning parameters. For this reason, we tend to focus on methods that _atuomatically select features_ during the normal process of training the model (as discussed below). 
```

Comment:
> can drastically increase the computational cost, especially if the model has many tuning parameters.

This is a multiplicative increase in computational cost correct? i get the sentiment but "drastically" feels like a exponential increase type of word.